enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,685 @@
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  creationTimestamp: "2024-06-11T14:22:37Z"
+  generation: 1
+  name: metallb-ovn-worker
+  namespace: metallb-system
+  resourceVersion: "1323"
+  uid: 99b64be3-4f36-4e0b-8704-75aa5182d89f
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 172.18.0.5
+        asn: 64512
+        disableMP: false
+        holdTime: 1m30s
+        keepaliveTime: 30s
+        passwordSecret: {}
+        port: 179
+        toAdvertise:
+          allowed:
+            mode: filtered
+        toReceive:
+          allowed:
+            mode: filtered
+  nodeSelector:
+    matchLabels:
+       kubernetes.io/hostname: ovn-worker
+```
+
+OVNKube-Controller will check that if this FRRConfiguration applies to its node, ovn-worker. For this example, a user-defined
+network named "blue", has been created with a network of 10.0.0.0/16, and a matching vrf exists in the Linux host. The slice
+of this supernet that has been allocated to node ovn-worker is 10.0.1./0/24. 
+
+##### Example 1: Advertising pod IPs from a user-defined network over BGP
+
+![](images/bgp_novpn_advertisements.png)
+
+In this example a user wants to expose network blue outside their OCP cluster so that pod IPs are reachable on the
+external network. The admin has creates the following RouteAdvertisements CR for the blue tenant:
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: RouteAdvertisements
+metadata:
+  name: default
+spec:
+  advertisements:
+    podNetwork: true
+  networkSelector:
+    matchLabels: 
+      k8s.ovn.org/metadata.name: blue
+  nodeSelector:
+   matchLabels:
+      kubernetes.io/hostname: ovn-worker
+```
+
+OVNKube-Controller will now see it needs to generate corresponding FRRConfiguration:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: route-advertisements-blue
+  namespace: metallb-system
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      vrf: blue
+      prefixes:
+        - 10.0.1.0/24
+    - asn: 64512
+      neighbors:
+         - address: 172.18.0.5
+           asn: 64512
+           toAdvertise:
+              allowed:
+                 prefixes:
+                    - 10.0.1.0/24
+  raw:
+    rawConfig: |-
+       router bgp 64512
+        address-family ipv4 unicast
+          import vrf blue
+          exit-address-family
+       router bgp 64512 vrf blue    
+        address-family ipv4 unicast
+          import vrf default
+          exit-address-family      
+  nodeSelector:
+     matchLabels:
+        kubernetes.io/hostname: ovn-worker
+```
+
+In the above configuration generated by OVN-Kubernetes, the subnet 10.0.1.0/24 which belongs to VRF blue, is being
+imported into the default VRF, and advertised to the 172.18.0.5 neighbor. This is because the targetVRF was defaulted so
+the routes are leaked and advertised in the default VRF. Additionally, routes are being imported from the default VRF
+into the blue VRF.
```

Comment:
we discussed this and it may still be useful for non-isolation reasons, but a user should be aware that by doing this they risk losing isolation to other networks also leaked into the default VRF. I'll make a note of that in the enhancement. Does that resolve the issue?

Reply:
Fine by me :-)

Reply:
updated and explained this better in the enhancement